kt/kt.go

```diff
@@ -191,9 +191,7 @@ func newConn(host string, port int, poolsize int, timeout time.Duration, creds s
 
 	// connectivity check so that we can bail out
 	// early instead of when we do the first operation.
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-	_, _, err = c.doRPC(ctx, "/rpc/void", nil)
+	err = CheckConn(c)
```

Comment:
c.CheckConn()?

Reply:
Fixed in 8ca3d0b14821dfae30e0d679e276433fc2d44ca9 that also sorted out couple other issues, such as reading the timeout from the struct.